### PR TITLE
Bullet featherstone: Add JointTransmittedWrench for F-T sensor

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🎉 New feature

Closes #<NUMBER>

## Summary

Adds support for joint transmitted wrench in bullet-featherstone.

Currently doesn't work...

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by nkoenig** [comment]
@mjcarroll, I think the use of SDF's `relative_to` in the `<pose>` tags could be the cause of the test failures. Writing this down in case it rings a bell with you.

**Comment by nkoenig** [comment]
In order for the tests to pass, we need the change in https://github.com/gazebo-forks/bullet3/pull/1, which would require us to roll our own bullet3 debian.

File: test/common_test/joint_transmitted_wrench_features.cc
Comment (scpeters) [inline test/common_test/joint_transmitted_wrench_features.cc:1]: If `JointTransmittedWrenchFixture` and its tests are moving to a separate file, we should have a clean pull request for the move and a separate pull request for modifications. It is very difficult to review the changes couple with a move of the code from one place to another

Comment (scpeters) [reply test/common_test/joint_transmitted_wrench_features.cc:1]: thanks!

File: test/common_test/joint_transmitted_wrench_features.cc [hunk 1]
```diff
@@ -431,5 +454,13 @@ TYPED_TEST(JointTransmittedWrenchFixture, ContactForces)
   EXPECT_TRUE(gz::physics::test::Equal(expectedWrenchAtMotorJointInJoint,
                                        wrenchAtMotorJointInJoint, 1e-4));
 }
+*/
```
Comment (scpeters) [inline test/common_test/joint_transmitted_wrench_features.cc:457]: I'd prefer not to disable this test for all physics engines if it passes for some of them

File: test/common_test/joint_transmitted_wrench_features.cc
Comment (scpeters) [reply test/common_test/joint_transmitted_wrench_features.cc:1]: I didn't notice the `main` function was missing from #495, but I've added it in https://github.com/gazebosim/gz-physics/pull/496 just to make sure we have a working baseline test for comparison

Comment (nkoenig) [reply test/common_test/joint_transmitted_wrench_features.cc:1]: Dohh, thanks.

**Comment by nkoenig** [comment]
@mjcarroll It looks like there is an collision issue. When running the `/test/common_test/worlds/pendulum_joint_wrench.sdf` with the `gz-physics-bullet-featherstone-plugin`, the pendulum arm will freely swing through the box. However, the box-box collision seems to work. Any ideas?

bullet-penetration

**Comment by nkoenig** [comment]
> @mjcarroll It looks like there is an collision issue. When running the `/test/common_test/worlds/pendulum_joint_wrench.sdf` with the `gz-physics-bullet-featherstone-plugin`, the pendulum arm will freely swing through the box. However, the box-box collision seems to work. Any ideas?
> 
> bullet-penetration [ bullet-penetration ](https://user-images.githubusercontent.com/1587438/230948725-d9f3c824-6993-47de-a2c5-9d6f20423569.gif) [ ](https://user-images.githubusercontent.com/1587438/230948725-d9f3c824-6993-47de-a2c5-9d6f20423569.gif)

I believe only the first collision within a model is actually created.

**Comment by mjcarroll** [comment]
> Any ideas?

Is it specific to the addition of this feature, or do you have the same problem with all bullet featherstone?

**Comment by nkoenig** [comment]
> > Any ideas?
> 
> Is it specific to the addition of this feature, or do you have the same problem with all bullet featherstone?

Looks like it's associated with all bullet featherstone, not just this PR.

**Comment by nkoenig** [comment]
[This `if`](https://github.com/gazebosim/gz-physics/blob/gz-physics6/bullet-featherstone/src/SDFFeatures.cc#L853) allows only first collider in a link to be created.

**Comment by nkoenig** [comment]
> [This `if`](https://github.com/gazebosim/gz-physics/blob/gz-physics6/bullet-featherstone/src/SDFFeatures.cc#L853) allows only first collider in a link to be created.

Fixed in https://github.com/gazebosim/gz-physics/pull/505

**Comment by nkoenig** [comment]
@mjcarroll I've updated the tests to skip when the bullet version is less than 3.25, which is the current version on github. See [23abcc3](https://github.com/gazebosim/gz-physics/pull/434/commits/23abcc3b3b6e026b027ca65fa0352b2973709555)

**Review by nkoenig — APPROVED** [review]
